File: app/frontend/src/app/game/components/2_navigasi_menu/2_navigasi_bawah/5_geopolitik/2_organisasi_internasional/1_organisasi_PBB/logic/python/membership_movement/2_leave_logic/leave_handler.py
import random

class LeaveHandler:
    """
    Handles the specific logic for an AI country to leave an organization.
    Ensures that leaving is rare (15% probability).
    """
    def __init__(self, exit_probability=0.15):
        self.exit_probability = exit_probability

    def evaluate_leave(self, country_name, decision_weight):
        """
        Returns True if the country decides to leave.
        """
        # Only consider leaving if decision weight is critically low
        if decision_weight < 20:
            # Enforce the rare exit probability
            if random.random() < self.exit_probability:
                return True
        return False

File: app/frontend/src/app/game/components/2_navigasi_menu/2_navigasi_bawah/5_geopolitik/2_organisasi_internasional/1_organisasi_PBB/logic/python/membership_movement/2_leave_logic/LeaveHandler.ts
/**
 * LeaveHandler TS Bridge: Matches the logic in leave_handler.py
 */
export class LeaveHandler {
    private exitProbability: number;

    constructor(exitProbability = 0.15) {
        this.exitProbability = exitProbability;
    }

    public evaluateLeave(countryName: string, decisionWeight: number): boolean {
        if (decisionWeight < 20) {
            return Math.random() < this.exitProbability;
        }
        return false;
    }
}
